src/main/scala/mesosphere/marathon/state/RootGroup.scala

```diff
@@ -242,7 +242,7 @@ class RootGroup(
     */
   def updateApp(
     appId: PathId, fn: Option[AppDefinition] => AppDefinition, version: Timestamp = Group.defaultVersion): RootGroup = {
-    val oldGroup = group(appId.parent).getOrElse(Group.empty(appId.parent))
+    val oldGroup = group(appId.parent.asAbsolutePath).getOrElse(Group.empty(appId.parent.asAbsolutePath))
```

Comment:
I think at some point we should change the signature on the Root group to only accept AbsolutePathIds for updateApp, updatePod, etc. Not the scope for this PR though

Reply:
Be my guest 🙂 I tried and found it way too much.